pkg/model/crd.go

```diff
@@ -170,17 +170,65 @@ func (r *CRD) SpecFieldNames() []string {
 // GoCodeSetInput returns the Go code that sets an input shape's member fields
 // from a CRD's Spec fields.
 //
-// We loop through the Spec fields, outputting code that looks something like
-// this:
+// Assume a CRD called Repository that looks like this pseudo-schema:
 //
-//   res.SetAttributes(r.ko.Spec.Attributes)
-//   res.SetName(*r.ko.Spec.Name)
-//   tmp0 := []*svcsdk.Tag{}
-//   res.Tags = tmp0
+// .Status
+//   .Authors ([]*string)
+//   .ImageData
+//     .Location (*string)
+//     .Tag (*string)
+//   .Name (*string)
+//
+// And assume an SDK Shape CreateRepositoryInput that looks like this
+// pseudo-schema:
+//
+// .Repository
+//   .Authors ([]*string)
+//   .ImageData
+//     .Location (*string)
+//     .Tag (*string)
+//   .Name
+//
+// This function is called from a template that generates the Go code that
+// represents linkage between the Kubernetes objects (CRs) and the aws-sdk-go
+// (SDK) objects. If we call this function with the following parameters:
+//
+//  opType:			OpTypeCreate
+//  sourceVarName:	ko.Spec
+//  targetVarName:	res
+//  indentLevel:	1
+//
+// Then this function should output something like this:
+//
+//   field1 := []*string{}
```

Comment:
nit: it should be "field0 := []*string{}" right?

Reply:
nice catch, eagle eyes! :)